opensafely/codelists.py

```diff
@@ -63,7 +68,32 @@ def main():
     pass
 
 
-def update(codelists_dir=None):
+def add(codelist_url, codelists_dir=None):
+    if not codelists_dir:
+        codelists_dir = Path.cwd() / CODELISTS_DIR
+
+    codelists_file = get_codelist_file(codelists_dir)
+
+    if OPENCODELISTS_BASE_URL not in codelist_url:
+        exit_with_error(f"Unable to parse URL, {OPENCODELISTS_BASE_URL} not found")
+    line = codelist_url.replace(f"{OPENCODELISTS_BASE_URL}/codelist/", "")
+
+    # trim any anchors from the end
+    if "#" in line:
+        line = line[:line.find("#")]
+
+    # turn a download url into a base codelist version url
+    if line.endswith(".csv"):
+        line = line[:line.rfind("/")]
+
+    with codelists_file.open("r+") as f:
+        if not f.readlines()[-1].endswith("\n"):
+            line = "\n" + line
+        f.write(line + "\n")
+    update(codelists_dir, delete_unknown=False)
```

Comment:
Calling `update(..)` here seems like it's going to cause some unwanted side effects. I can see you've guarded against one unintended side effect by changing the behaviour of `update(..)` when called by `add(..)`, but I'm not sure that users would expect or want their other codelists to change just because they've added an extra one.

Could you move the relevant code out of `update(..)` into a separate function, which is called by both `update(..)` and `add(..)` instead? This would remove the need to change the behaviour of `update(..)`.